ui/src/BaseEditor.ts

```diff
@@ -0,0 +1,169 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { DialogResponses } from './DialogResponses' ;
+import { UserCancelledError } from './errors';
+import { localize } from "./localize";
+import { TemporaryFile } from './utils/TemporaryFile';
+
+// tslint:disable-next-line:no-unsafe-any
+export abstract class BaseEditor<ContextT> implements vscode.Disposable {
+    private fileMap: { [key: string]: [vscode.TextDocument, ContextT] } = {};
+    private ignoreSave: boolean = false;
+
+    constructor(readonly showSavePromptKey: string, readonly outputChanel?: vscode.OutputChannel) {
+    }
+
+    public abstract getData(context: ContextT): Promise<string>;
+    public abstract updateData(context: ContextT, data: string): Promise<string>;
+    public abstract getFilename(context: ContextT): Promise<string>;
+    public abstract getSize(context: ContextT): Promise<number>;
+    public abstract getSaveConfirmationText(context: ContextT): Promise<string>;
+
+    public async showEditor(context: ContextT): Promise<void> {
+        const size: number = await this.getSize(context);
+        const fileName: string = await this.getFilename(context);
+        const splitFileName: string[] = fileName.split('.');
+        const extension: string = splitFileName[splitFileName.length - 1];
+
+        this.appendToOutput(localize('opening', 'Opening "{0}" ...', fileName));
+        if (size > 50 /*Megabytes*/) {
+            const message: string = localize('tooLargeError', '"{0}" is too large to download.', fileName);
+
+            await vscode.window.showWarningMessage(message, DialogResponses.OK);
+            this.appendLineToOutput(localize('failed', "Failed."));
+            this.appendLineToOutput(localize('errorDetails', 'Error Details: {0}', message));
+        } else if (extension === 'exe' || extension === 'img' || extension === 'zip') {

```

Comment:
Again - why are we hard-coding file extensions here? Wouldn't it be better to let Azure determine what can or cannot be downloaded?

(Again - cosmos will be different - it's always JSON files)

Reply:
I believe this is also a text editor limitation.  It wouldn't make sense to allow any extension to attempt to open an exe, img, or zip file in the editor.

Reply:
VS Code already has logic for files it cannot display:
<img width="761" alt="screen shot 2017-12-11 at 5 02 01 pm" src="https://user-images.githubusercontent.com/11282622/33861940-27e4740e-de95-11e7-8f45-be56293a6a0c.png">

I just don't think we should have our own custom logic for this. At the end of the day it's not _our_ limitation - it's either Azure's or VS Code's.